src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/Tensor.cs

```diff
@@ -2064,98 +2064,147 @@ public static ref readonly TensorSpan<T> StackAlongDimension<T>(scoped ReadOnlyS
+            {
+                TensorShape tmpShape = TensorShape.Create(tensor.Lengths[1..], tensor.Strides[1..], tensor.IsPinned);
 
-                    if (length != tensor.Lengths[0])
-                    {
-                        sb.Append(separator);
-                        sb.Append("...");
-                    }
+                ReadOnlyTensorSpan<T> tmpTensor = new ReadOnlyTensorSpan<T>(ref tensor._reference, tmpShape);
+                ToString(tmpTensor, maximumLengths[1..], sb, indentLevel + 1);
 
-                    sb.Append(separator);
+                for (nint i = 1; i < length; i++)
+                {
+                    sb.AppendLine(",");
+                    tmpTensor = new ReadOnlyTensorSpan<T>(ref Unsafe.Add(ref tensor._reference, i * tensor.Strides[0]), tmpShape);
+                    ToString(tmpTensor, maximumLengths[1..], sb, indentLevel + 1);
                 }
+
+                if (length != tensor.Lengths[0])
+                {
+                    sb.AppendLine(",");
+                    sb.Append(' ', (indentLevel + 1) * 2);
+                    sb.Append("..");
+                }
+                sb.AppendLine();
             }
+            else
+            {
+                sb.Append(' ', (indentLevel + 1) * 2);
+                sb.AppendLine("..");
+            }
+            sb.Append(' ', indentLevel * 2);
+
+            if (indentLevel != 0)
+            {
+                sb.Append(']');
+            }
+        }
+
+        private static void ToString<T>(in ReadOnlyTensorSpan<T> tensor, nint length, StringBuilder sb, int indentLevel)
+        {
+            sb.Append(' ', indentLevel * 2);
+            sb.Append('[');
+
+            if (length != 0)
+            {
+                sb.Append(tensor._reference);
+
+                for (nint i = 1; i < length; i++)
+                {
+                    sb.Append(", ");
+                    sb.Append(Unsafe.Add(ref tensor._reference, i));
+                }
+
+                if (length != tensor.Lengths[0])
+                {
+                    sb.Append(", ..");
+                }
+            }
+            else
+            {
+                sb.Append("..");
+            }
+
             sb.Append(']');
         }
 
-        /// <summary>
-        /// Creates a <see cref="string"/> representation of the <see cref="Tensor{T}"/>."/>
-        /// </summary>
-        /// <param name="tensor">The <see cref="Span{T}"/> you want to represent as a string.</param>
-        /// <param name="maximumLengths">Maximum Length of each dimension</param>
-        /// <returns>A <see cref="string"/> representation of the <paramref name="tensor"/></returns>
-        public static string ToString<T>(this Tensor<T> tensor, ReadOnlySpan<nint> maximumLengths)
-            => tensor.AsReadOnlyTensorSpan().ToString(maximumLengths);
+        private static StringBuilder AppendJoin<T>(this StringBuilder sb, string separator, ReadOnlySpan<T> values)
+        {
```

Comment:
isn't the separator always `", "`? I am seeing we are hardcoding it in other places why we are passing it here? should we define it as const string?